examples/transports/README.md

```diff
@@ -243,58 +195,58 @@ As a rule, a libp2p node will only be capable of using a transport if: a) it has
 Let's update our flow to create nodes and see how they behave when dialing to each other:
 
 ```JavaScript
-parallel([
-  (cb) => createNode('/ip4/0.0.0.0/tcp/0', cb),
-  // Here we add an extra multiaddr that has a /ws at the end, this means that we want
-  // to create a TCP socket, but mount it as WebSockets instead.
-  (cb) => createNode(['/ip4/0.0.0.0/tcp/0', '/ip4/127.0.0.1/tcp/10000/ws'], cb),
-  (cb) => createNode('/ip4/127.0.0.1/tcp/20000/ws', cb)
-], (err, nodes) => {
-  if (err) { throw err }
-
-  const node1 = nodes[0]
-  const node2 = nodes[1]
-  const node3 = nodes[2]
-
-  printAddrs(node1, '1')
-  printAddrs(node2, '2')
-  printAddrs(node3, '3')
-
-  node1.handle('/print', print)
-  node2.handle('/print', print)
-  node3.handle('/print', print)
-
-  // node 1 (TCP) dials to node 2 (TCP+WebSockets)
-  node1.dialProtocol(node2.peerInfo, '/print', (err, conn) => {
-    if (err) { throw err }
-
-    pull(pull.values(['node 1 dialed to node 2 successfully']), conn)
-  })
-
-  // node 2 (TCP+WebSockets) dials to node 2 (WebSockets)
-  node2.dialProtocol(node3.peerInfo, '/print', (err, conn) => {
-    if (err) { throw err }
-
-    pull(pull.values(['node 2 dialed to node 3 successfully']), conn)
-  })
-
-  // node 3 (WebSockets) attempts to dial to node 1 (TCP)
-  node3.dialProtocol(node1.peerInfo, '/print', (err, conn) => {
-    if (err) {
-      console.log('node 3 failed to dial to node 1 with:', err.message)
-    }
-  })
-})
+const [peerInfo1, peerInfo2, peerInfo3] = await Promise.all([
+  PeerInfo.create(),
+  PeerInfo.create(),
+  PeerInfo.create()
+])
+const [node1, node2, node3] = await Promise.all([
+  createNode(peerInfo1, '/ip4/0.0.0.0/tcp/0'),
+  createNode(peerInfo2, ['/ip4/0.0.0.0/tcp/0', '/ip4/127.0.0.1/tcp/10000/ws']),
+  createNode(peerInfo3, '/ip4/127.0.0.1/tcp/20000/ws')
+])
+
+printAddrs(node1, '1')
+printAddrs(node2, '2')
+printAddrs(node3, '3')
+
+node1.handle('/print', print)
+node2.handle('/print', print)
+node3.handle('/print', print)
+
+// node 1 (TCP) dials to node 2 (TCP+WebSockets)
+const { stream } = await node1.dialProtocol(node2.peerInfo, '/print')
+await pipe(
+  ['node 1 dialed to node 2 successfully'],
+  stream
+)
+
+// node 2 (TCP+WebSockets) dials to node 2 (WebSockets)
+const { stream: stream2 } = await node2.dialProtocol(node3.peerInfo, '/print')
+await pipe(
+  ['node 2 dialed to node 3 successfully'],
+  stream2
+)
+
+// node 3 (WebSockets) attempts to dial to node 1 (TCP)
+try {
+  await node3.dialProtocol(node1.peerInfo, '/print')
+} catch (err) {
+  console.log('node 3 failed to dial to node 1 with:', err.message)
+}
```

Comment:
The README no longer matches up with the file in 3. The readme example attempts to show a failure dialing from node 3 to node 1. This previously failed because we would remove transports we didn't have addresses for, which is unnecessary. Since we didn't have a TCP transport to dial to node 1 from 3, the dial would fail. Now that node3 has the transport it passes now. To properly demonstrate the failure we should create node 3 without a TCP transport, and node 1 without a websocket transport.

Reply:
changed it back to fail